docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,444 @@ overview page.
+in our tools.
+
+Fluentd (optional)
+""""""""""""""""""
+
+Fluentd could be optionally added between Fluent Bit and Loki to offer
+more plugins and reduce the impact of Fluent Bit on workload nodes.
+
+It's not recommended for small platforms (few nodes) as it adds a big overhead
+in resources consumption.
+
+Rejected Design Choices
+-----------------------
+
+Promtail + Loki
+~~~~~~~~~~~~~~~
+
+Promtail has been rejected because, even if it consumes very few resources,
+it can only integrate with Loki and we need something more versatile, with
+the ability to interact with different distributors.
+
+Fluentd + Loki
+~~~~~~~~~~~~~~
+
+This architecture has been rejected because it means we need 1 fluentd instance
+per node, which increases a lot the resources consumption compared to other
+solutions.
+
+Logstash + Elasticsearch
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This architecture is probably the most common one, but it has not been taken
+into consideration because we want to focus on having the minimum resources
+consumption and these components can really hog RAM & CPU. Beats could be used
+as the log collector to reduce the impact of Logstash, but Elasticsearch still
+consumes a lot of resources.
+
+Implementation Details
+----------------------
+
+Deployment
+~~~~~~~~~~
+
+All the components will be deployed using Kubernetes manifests through Salt
+inside a ``metalk8s-logging`` Namespace.
+
+As we already did for other components (e.g. Prometheus Operator), the
+manifests for Loki and Fluent Bit will be generated statically by rendering
+the `loki-stack helm chart`_::
+
+  helm repo add loki https://grafana.github.io/loki/charts
+  helm repo update
+  helm fetch -d charts --untar loki/loki-stack
+  ./charts/render.py loki-stack --namespace metalk8s-logging \
+    charts/loki-stack.yaml charts/loki-stack/ \
+    > salt/metalk8s/addons/loki-stack/deployed/charts.sls
+
+.. _loki-stack helm chart: https://github.com/grafana/loki/tree/master/production/helm
+
+We will use the same approach for Fluentd, using the `fluentd helm chart`_::
+
+  helm fetch -d charts --untar stable/fluentd
+  ./charts/render.py fluentd --namespace metalk8s-logging charts/fluentd.yaml \
+    charts/fluentd/ > salt/metalk8s/addons/fluentd/deployed/charts.sls
+
+.. _fluentd helm chart: https://hub.helm.sh/charts/stable/fluentd
+
+Container images must be added in the buildchain
+``buildchain/buildchain/image.py`` and then replaced in
+``charts/loki-stack.yaml`` and ``charts/fluentd.yaml`` using the
+``__image__()`` macro to be replaced by jinja syntax when rendering the
+charts and then dynamically computed at run time.
+
+Fluent Bit
+""""""""""
+
+Fluent Bit will be deployed as a DaemonSet, because we need one collector on
+each control or workload plane node to be able to collect logs from both the
+Kubernetes platform and the applications that run on top of it.
+
+Loki
+""""
+
+Loki will be deployed as a StatefulSet as it needs persistent volumes to write
```

Comment:
Not possible to split components using filesystem as storage backend, which means no autoscaling either because the monolithic pod is stateful because of the ingester volume.

Reply:
Can't split and horizontally scale - true
Can't autoscale - debatable: we may want to adjust `memory` and `cpu` consumption from the querier container (through vertical **autoscaling** :wink:)